### ISSUE TITLE
No difference eos5D & eos5Dmk2 in central view

### ISSUE THREAD

**Issue by dim162** [darktable-org/darktable#7725]
**Describe the bug**
I have raw from canon eos 5D & eos 5D mk2. After choosing the rules "camera" in "collect images", i have the list with Canon eos 5D & 5D mk2, with the number of each. When i select canon eos 5D mk2, these are shown in the central view. When i select canon eos 5D, the canon eos 5D + canon eos 5D mk2 are shown. The number shown on the top of the central view is the addition of the 2 types of raw.

**To Reproduce**
Raw from canon eos 5D & 5D mk2 needed

**Expected behavior**
If the rule make the difference, i expect the same difference when it is exposed in the central view

**Screenshots**
![Screenshot_2021-01-07_06-07-08](https://user-images.githubusercontent.com/10468352/103853555-a5bb9e00-50ae-11eb-9c95-7a910ebdfa38.png)
![Screenshot_2021-01-07_06-05-10](https://user-images.githubusercontent.com/10468352/103853558-a6543480-50ae-11eb-91a2-1c09821bfb07.png)

**Platform (please complete the following information):**
 - Darktable Version: 3.2.1, 3.4, 3.5
 - Ubuntu 19.10
 - OpenCL activated 
 - nvidia geforce 1050ti+760, nvidia ge force 1050ti +quadro rtx 4000, driver 440.100

**Additional context**

 - Can you reproduce with another Darktable version? yes
 - Can you reproduce with a RAW or Jpeg or both?i have only raw
 - Are the steps above reproduce with a fresh edit (removing history)?yes
 - Dt 3.4,3.5 from Suse, Dt 3.2.1 compiled
 - Is the issue still present using an empty/new config-dir? yes

**Comment by johnny-bit** [comment]
It's most likely caused because collection builds query as "[Camera Name][wildcard]" which means asking for "Camera model" sees all "Camera model rev xx"...

I think i remember there was issue recently with similar problem: PR https://github.com/darktable-org/darktable/pull/7333 by @tomaszg7 fixed https://github.com/darktable-org/darktable/issues/6494 - I think similar thing could and should be done for collections

**Comment by parafin** [comment]
It's a duplicate of #4527 (very long-standing issue).

**Comment by johnny-bit** [comment]
Thanks @parafin ! I knew I saw something very similar, but all I found was @tomaszg7's think with basecurve presets ;)

**Comment by HansBull** [comment]
> It's most likely caused because collection builds query as "[Camera Name][wildcard]" which means asking for "Camera model" sees all "Camera model rev xx"...

Does it really?

**Comment by phweyland** [comment]
Is somebody working on this ? if not, it looks easy enough for I take it. :)

**Comment by johnny-bit** [comment]
Go ahead!

**Comment by phweyland** [comment]
@dim162 , could you share 1 raw file of each camera please?
That will help me to find out where is the issue and check the solution if I find one. :)
Thanks in advance!

**Comment by johnny-bit** [comment]
@phweyland - grab couple from https://raw.pixls.us/ :) there are samples for each camera ;)

**Comment by phweyland** [comment]
In collect, the list of cameras considers all combinations of (maker, model) found on images with their related count.
When selecting an entry, lighttable shows all combinations of (maker, model) where the selected entry(1) does appear. It's like @johnny-bit, but the code is just a bit more complex than a wildcard, which was intentional was intentional... Maybe the equivalent on collect side has been removed at some point of the time.
(1) User wildcard on entry doesn't work (while it works for lens for example).
 
In short. The goal is to show to the user matching numbers. Two options: which one do we want to keep ?

**Comment by phweyland** [comment]
Humm... Not sure my question does make sense. I'll start keeping the collect behavior, make lighttable match with it and make user wildcard work....

**Comment by ptilopteri** [comment]
the sorting will work as dt is presently configured
specify on the first collect images 
    include eos5D
and "narrow down search"
    exclude eos5Dmk2

and it is possible to make a preset

**Comment by HansBull** [comment]
change line 1391 in collection.c to

    if (!needle || g_strcmp0(haystack, needle) == 0)
